homeassistant/components/zha/select.py

```diff
@@ -408,6 +410,66 @@ class AqaraApproachDistance(ZCLEnumSelectEntity):
     _attr_translation_key: str = "approach_distance"
 
 
+@CONFIG_DIAGNOSTIC_MATCH(
+    cluster_handler_names="opple_cluster", models={"lumi.magnet.ac01"}
+)
+class AqaraMagnetAC01DetectionDistance(ZCLEnumSelectEntity):
+    """Representation of a ZHA detection distance configuration entity."""
+
+    _unique_id_suffix = "detection_distance"
+    _attribute_name = "detection_distance"
+    _enum = MagnetAC01OppleCluster.DetectionDistance
+    _attr_translation_key: str = "detection_distance"
+
+
+@CONFIG_DIAGNOSTIC_MATCH(
+    cluster_handler_names="opple_cluster", models={"lumi.switch.acn047"}
+)
+class AqaraT2RelaySwitchMode(ZCLEnumSelectEntity):
+    """Representation of a ZHA switch mode configuration entity."""
+
+    _unique_id_suffix = "switch_mode"
+    _attribute_name = "switch_mode"
+    _enum = T2RelayOppleCluster.SwitchMode
+    _attr_translation_key: str = "switch_mode"
+
+
+@CONFIG_DIAGNOSTIC_MATCH(
+    cluster_handler_names="opple_cluster", models={"lumi.switch.acn047"}
+)
+class AqaraT2RelaySwitchType(ZCLEnumSelectEntity):
+    """Representation of a ZHA switch type configuration entity."""
+
+    _unique_id_suffix = "switch_type"
+    _attribute_name = "switch_type"
+    _enum = T2RelayOppleCluster.SwitchType
+    _attr_translation_key: str = "switch_type"
+
+
+@CONFIG_DIAGNOSTIC_MATCH(
+    cluster_handler_names="opple_cluster", models={"lumi.switch.acn047"}
+)
+class AqaraT2RelayStartupOnOff(ZCLEnumSelectEntity):
+    """Representation of a ZHA startup on off configuration entity."""
+
+    _unique_id_suffix = "startup_on_off"
+    _attribute_name = "startup_on_off"
+    _enum = T2RelayOppleCluster.StartupOnOff
+    _attr_translation_key: str = "startup_on_off"
```

Comment:
We could probably use the existing `start_up_on_off` key here instead of creating a new one(?)

Reply:
good catch